qutebrowser/browser/commands.py

```diff
@@ -1510,3 +1510,29 @@ def debug_webaction(self, action, count=1):
         view = self._current_widget()
         for _ in range(count):
             view.triggerPageAction(member)
+
+    @cmdutils.register(instance='command-dispatcher', scope='window',
+                       maxsplit=0, no_cmd_split=True)
+    def jseval(self, js_code):
+        """Evaluate a JavaScript string.
+
+        Args:
+            js_code: The string to evaluate.
+        """
+        out = self._current_widget().page().mainFrame().evaluateJavaScript(
+                    js_code)
+
+        if out is None:
+            # Getting the actual error (if any) seems to be difficult. The
+            # error does end up in BrowserPage.javaScriptConsoleMessage(), but
+            # distinguishing between :jseval errors and errors from the webpage
+            # is not trivial...
+            message.info(self._win_id, 'No output or error')
+        else:
+            # The output can be a string, number, dict, array, etc. But *don't*
+            # output too much data, as this will make qutebrowser hang
+            out = str(out)
+            if len(out) > 5000:
+                message.info(self._win_id, out[:5000] + ' [...trimmed...]')
+            else:
+                message.info(self._win_id, out)
```

Comment:
I suggest using `utils.compact_text()` or `utils.elide()` here - `compact_text` will also compact newlines (I think that's good thing).